rust/xaynet-analytics/src/database/instance.rs

```diff
@@ -0,0 +1,116 @@
+use isar_core::{
+    collection::IsarCollection,
+    error::IsarError,
+    instance::IsarInstance,
+    object::{object_builder::ObjectBuilder, object_id::ObjectId},
+    schema::{collection_schema::CollectionSchema, Schema},
+};
+use std::vec::IntoIter;
+
+use crate::database::{
+    analytics_event::data_model::AnalyticsEvent,
+    common::{FieldProperty, IsarAdapter},
+};
+
+pub struct DbInstance {
+    instance: IsarInstance,
+}
+
+impl DbInstance {
+    const MAX_SIZE: usize = 10000000;
+    const ANALYTICS_EVENT_NAME: &'static str = "analytics_events";
+
+    pub fn new(path: &str) -> Result<Self, IsarError> {
+        match IsarInstance::create(path, Self::MAX_SIZE, Self::get_schema()) {
+            Ok(instance) => Ok(Self { instance }),
+            Err(e) => Err(e),
+        }
+    }
+
+    pub fn instance(&self) -> &IsarInstance {
```

Comment:
Why do we need to expose the instance?

Reply:
it's debris from a previous implementation, i forgot to clean it up